File: BINDINGS/cxx/eflxx_examples/src/elementaryxx/full/test_layout.cpp
#include "test.h"



void test_layout (void *data, Evas_Object *obj, void *event_info)
{
  Window *win = Window::factory ("layout", ELM_WIN_BASIC);
  win->setTitle ("Layout");
  win->setAutoDel (true);

  Background *bg = Background::factory (*win);
  win->addObjectResize (*bg);
  bg->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  bg->show ();

  Layout *ly = Layout::factory (*win);
  ly->setFile (searchEdjeFile ("elementaryxx-test.edj"), "layout");
  ly->setWeightHintSize (EVAS_HINT_EXPAND, EVAS_HINT_EXPAND);
  win->addObjectResize (*ly);
  ly->show ();

  Button *bt = Button::factory (*win);
  bt->setLabel ("Button 1");
  ly->setContent ("element1", *bt);
  bt->show ();

  Button *bt2 = Button::factory (*win);
  bt2->setLabel ("Button 2");
  ly->setContent ("element2", *bt2);
  bt2->show ();

  Button *bt3 = Button::factory (*win);
  bt3->setLabel ("Button 3");
  ly->setContent ("element3", *bt3);
  bt3->show ();

  win->show ();
}

File: BINDINGS/python/python-elementary/tests/test_layout.py
#!/usr/bin/env python
# encoding: utf-8
import os
import elementary
import edje
import ecore
import evas

#----- Layout -{{{-
def _event(*args, **kargs):
    print((args, kargs))

def layout_clicked(obj):
    win = elementary.Window("layout", elementary.ELM_WIN_BASIC)
    win.title_set("Layout")
    win.elm_event_callback_add(_event)
    win.autodel_set(True)
    if obj is None:
        win.callback_delete_request_add(lambda o: elementary.exit())

    bg = elementary.Background(win)
    win.resize_object_add(bg)
    bg.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
    bg.show()

    ly = elementary.Layout(win)
    ly.file_set("test.edj", "layout")
    ly.size_hint_weight_set(evas.EVAS_HINT_EXPAND, evas.EVAS_HINT_EXPAND)
    win.resize_object_add(ly)
    ly.show()

    bt = elementary.Button(win)
    bt.text_set("Button 1")
    ly.part_content_set("element1", bt)
    bt.elm_event_callback_add(_event)
    bt.elm_event_callback_del(_event)
    bt.show()

    bt = elementary.Button(win)
    bt.text_set("Button 2")
    ly.part_content_set("element2", bt)
    bt.show()

    bt = elementary.Button(win)
    bt.text_set("Button 3")
    ly.part_content_set("element3", bt)
    bt.show()

    win.show()
# }}}

#----- Main -{{{-
if __name__ == "__main__":
    elementary.init()

    layout_clicked(None)

    elementary.run()
    elementary.shutdown()
# }}}
# vim:foldmethod=marker
